tls/s2n_handshake_io.c

```diff
@@ -374,6 +360,15 @@ static message_type_t tls13_handshakes[S2N_HANDSHAKES_COUNT][S2N_MAX_HANDSHAKE_L
             CLIENT_CHANGE_CIPHER_SPEC, CLIENT_FINISHED,
             APPLICATION_DATA
     },
+
+    [NEGOTIATED | HELLO_RETRY_REQUEST | FULL_HANDSHAKE] = {
+            CLIENT_HELLO,
+            HELLO_RETRY_MSG,
+            CLIENT_HELLO,
+            SERVER_HELLO, SERVER_CHANGE_CIPHER_SPEC, ENCRYPTED_EXTENSIONS, SERVER_CERT, SERVER_CERT_VERIFY, SERVER_FINISHED,
+            CLIENT_CHANGE_CIPHER_SPEC, CLIENT_FINISHED,
+            APPLICATION_DATA
```

Comment:
Check your CHANGE_CIPHER_SPEC messages. I believe (as came up in the SAW proofs) that the CLIENT_CHANGE_CIPHER_SPEC needs to appear before the second ClientHello. RFC: https://tools.ietf.org/html/rfc8446#appendix-D.4

Reply:
I think it looks right now. Do we have the ability to test middlebox compat using any tooling?